keras_nlp/utils/text_generation_test.py

```diff
@@ -113,6 +113,46 @@ def token_probability_fn(inputs):
         expected_outputs = tf.concat([inputs, expected_outputs], axis=1)
         self.assertAllEqual(outputs, expected_outputs)
 
+    def test_model_compile(self):
+        def token_probability_fn(inputs):
+            prob = tf.constant([[0.01, 0.01, 0.08, 0.9]])
+            return tf.repeat(prob, 2, axis=0)
+
+        max_length = 5
+
+        class TestModel(tf.keras.Model):
+            def call(self, inputs, training=False):
+                if not training:
+                    generated = greedy_search(
+                        token_probability_fn,
+                        inputs,
+                        max_length=max_length,
+                        end_token_id=2,
+                        pad_token_id=0,
+                    )
+                    return generated
+                else:
+                    return inputs
+
+        def dummy_metric(y_true, y_pred):
+            if tf.reduce_all(tf.equal(y_true, y_pred)):
+                return 1
+            else:
+                return 0
+
+        inputs = tf.constant([[0, 1], [1, 2]])
+        model = TestModel()
+        model.compile(metrics=[dummy_metric])
```

Comment:
If you add `jit_compile=True` does the test still pass?

If yes, we should test this with both jit_compile=True and False, using https://docs.pytest.org/en/6.2.x/parametrize.html

If no, we should either try to fix things with jit_compilation, or make sure we track that on a follow up issue.

Reply:
@abheesht17 Let's try add a test case for `jit_compile=True`, and we can run it on GPU. We recently add GPU test support in this repo.

Reply:
It's not working with `jit_compile = True`. Complete error logs: https://p.ip.fi/2TNt.

Looks like it won't work with `shape_invariants`.